crates/starknet-os/src/hints/execution.rs

```diff
@@ -1119,6 +1121,21 @@ pub fn check_new_deploy_response(
     _constants: &HashMap<String, Felt252>,
 ) -> Result<(), HintError> {
     let response_ptr = get_ptr_from_var_name(vars::ids::RESPONSE, vm, ids_data, ap_tracking)?;
+
+    // Bugfix:
+    // If constructor_retdata_start is a Int(0) instead of a Relocatable, we need to do nothing. This
+    // can happen sometimes when a constructor is called but not defined in the class'es entry
+    // points. Immediately following this, `add_relocation_rule()` will be called with src=0, dest=0
+    // and it will also no-op.
+    let retdata_start_key: Relocatable =
+        (response_ptr + new_syscalls::DeployResponse::constructor_retdata_start_offset())?;
+    let maybe_retdata_start = vm.get_maybe(&retdata_start_key).unwrap();
+    let zero = MaybeRelocatable::Int(Felt252::ZERO);
+    if maybe_retdata_start == zero {
+        log::warn!("retdata_start is 0, not a relocatable. doing nothing");
+        return Ok(());
+    }
```

Comment:
@notlesh could you add more context to this? 
It feels weird that in a check function, we got something that we don not expect and we do nothing.

Ideally you add a link to the code that explains why this "error bypass" is correct

Reply:
Yeah, it is a weird place for part of this hack. I elaborated, see [3ee3f5d](https://github.com/keep-starknet-strange/snos/pull/393/commits/3ee3f5df5e217a281b9cfd1572759e44a6a92b77)